src/Components/test/testassets/Components.TestServer/RazorComponents/Pages/StreamingRendering/StreamingSetNotFound.razor

```diff
@@ -0,0 +1,12 @@
+@page "/streaming-set-not-found"
+@attribute [StreamRendering]
+@inject NavigationManager NavigationManager
+
+@code{
+    protected override async Task OnInitializedAsync()
+    {
+        // Simulate some delay before triggering NotFound to start streaming response
+        await Task.Delay(1000);
```

Comment:
```suggestion
        await Task.Yield();
```
  
  If you need to check state before/after streaming rendering starts for some reason, TaskCompletionSource is the more correct alternative to get 100 reliability (we have some tests that do this, even though it's a bit cumbersome to setup)